src/System.IO/src/System/IO/BinaryWriter.cs

```diff
@@ -71,6 +72,8 @@ protected virtual void Dispose(bool disposing)
         {
             if (disposing)
             {
+                ArrayPool<byte>.Shared.Return(_buffer);
```

Comment:
This should return in the finalization path too. Speaking of which, should we add finalizers to BinaryReader and BinaryWriter now that they use pooled buffers? 

cc: @vancem, @stephentoub